### PR TITLE
DO-947, DENG-784 Rollups and add topsites metrics for Newtab interactions 

### PR DESCRIPTION
**Pull request by ANich**
RE: https://mozilla-hub.atlassian.net/browse/DO-947, https://mozilla-hub.atlassian.net/browse/DENG-784

Introducing some pre-aggregations for newtab Looker performance. I did a best effort at picking dimensions and measures that are used in dashboards, tagging folks for a second opinion.

----

Checklist for reviewer:

When adding a new derived dataset:
- [ ] Ensure that the data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data may be available in [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).
- [ ] Avoid merging a PR that includes the logic of a [core metric](https://docs.telemetry.mozilla.org/metrics/index.html) or complex business logic. The recommendation is to implement core business logic in bigquery-etl. E.g. The [type of search](https://github.com/mozilla/bigquery-etl/blob/a3e59f90326816a2ecaaa3e9d5b57fe9552f7d70/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L781) or the [calculation of DAU or visited URIs](https://github.com/mozilla/bigquery-etl/blob/9bca48821a8a0d40b1700cc14ecd8068d132ed06/sql/moz-fx-data-shared-prod/telemetry_derived/firefox_desktop_exact_mau28_by_dimensions_v1/query.sql).
- [ ] Avoid merging code in Looker Explores/Views that implement analysis with multiple lines of code or that will be likely replicated in the future. Instead, aim for extending an existing dataset to include the required logic, and use [Looker aggregates](https://cloud.google.com/looker/docs/aggregate_awareness) to facilitate the analysis.
- [ ] Avoid merging a PR with logic that requires validation and health checks. It is recommended to implement it in bigquery-etl for full test coverage and failure alerts.

### CODE REVIEW

**Comment by irrationalagent** [comment]
I'll defer to @mmccorks on the dimensions / measures question. This seems to cover the ones i used in my most recent dashboard.

Also, for validation's sake, when we run a query through the explore once this change merges, is there an easy way to see from the UI that the rollup (rather than the source table) was used? Does it just show up in the SQL tab like `SELECT blah FROM rollup__newtab_interactions_topsites_per_day`?

Finally, out of curiosity, can the aggregate awareness stuff work with tables that are defined separately from looker, e.g. in BQ ETL? I.e. could I in theory tell looker, "this table that already exists is a rollup of this explore, use that when you can?" (I am NOT saying that is what we should do here, I'm just trying to understand what's possible).

**Comment by ANich** [comment]
> Does it just show up in the SQL tab like `SELECT blah FROM rollup__newtab_interactions_topsites_per_day`?

Yup, it shows up in the SQL tab and also at the top-right where the amount of data scanned shows up. (Scanned x instead of y using aggregate table)


> Finally, out of curiosity, can the aggregate awareness stuff work with tables that are defined separately from looker, e.g. in BQ ETL? I.e. could I in theory tell looker, "this table that already exists is a rollup of this explore, use that when you can?" (I am NOT saying that is what we should do here, I'm just trying to understand what's possible).

Not that I'm aware of, though it has been a [feature request](https://portal.feedback.us.pendo.io/app/#/case/64941?currentProductId=53ee6715-9dcb-471f-bd41-7846ac4d9e41&section=requests). The closest we could get to this is using something like materialized views since bigquery can automatically rewrite your query when it matches but there's some challenges there as well

File: firefox_desktop/explores/newtab_interactions.explore.lkml
Comment (mmccorks) [inline firefox_desktop/explores/newtab_interactions.explore.lkml:182]: Can we also filter this to only countries with Topsites enabled?

File: firefox_desktop/explores/newtab_interactions.explore.lkml
Comment (mmccorks) [inline firefox_desktop/explores/newtab_interactions.explore.lkml:94]: Can we also also add a filter here to only include countries with Pocket?

**Review by mmccorks — APPROVED** [review]
Looks good to me, I suggested adding filters for the topsites and pocket countries but it's not a blocker.

**Comment by ANich** [comment]
Pausing this until https://github.com/mozilla/bigquery-etl/pull/3581 and related issues are resolved

**Comment by ANich** [comment]
I paused this anticipating a few column additions so we could do (only) one pre-aggregation run. I'll re-configure this for the columns added in https://github.com/mozilla/bigquery-etl/pull/3702 and merge. When columns need to be added at a later point we' can modify this

File: firefox_desktop/explores/newtab_interactions.explore.lkml
Comment (ANich) [reply firefox_desktop/explores/newtab_interactions.explore.lkml:172]: We don't currently have a list of those iirc

File: firefox_desktop/explores/newtab_interactions.explore.lkml
Comment (irrationalagent) [reply firefox_desktop/explores/newtab_interactions.explore.lkml:95]: +1 if its possible

File: firefox_desktop/explores/newtab_interactions.explore.lkml
Comment (irrationalagent) [inline firefox_desktop/explores/newtab_interactions.explore.lkml:145]: just for the benefit of my understanding, where does this live, or how is it generated etc?

Comment (ANich) [reply firefox_desktop/explores/newtab_interactions.explore.lkml:145]: These are generated in looker-hub by looking at a table's last updated timestamp (usually by ETL). Here's the most relevant doc: https://docs.telemetry.mozilla.org/cookbooks/looker/performance_caching.html, though in the caching context, and the [proposal](https://docs.google.com/document/d/1dPwa5KZnKHzYj3hULHy9unczCXugwr6rZHW1gveQ2fc/edit#heading=h.5x0d5h95i329).

File: firefox_desktop/explores/newtab_interactions.explore.lkml
Comment (ANich) [inline firefox_desktop/explores/newtab_interactions.explore.lkml:41]: I'll keep an eye on this. The documentation [here](https://cloud.google.com/looker/docs/aggregate_awareness#approximate_distinct) mentions that distinct counts are only pre-aggregated if all the dimensions are included but an exception is approximate distincts. These are all approximate distincts but the documentation is unclear on whether this will be pre-aggregated in an example say only using one or two of these dimensions.

Comment (ANich) [reply firefox_desktop/explores/newtab_interactions.explore.lkml:41]: This doesn't seem to be working, will follow up in https://mozilla-hub.atlassian.net/browse/DS-2592

**Pull request merged**
